core/src/main/java/com/schemaapp/core/services/impl/CDNDataAPIServiceImpl.java

```diff
@@ -110,9 +113,13 @@ private void processPage(ResourceResolver resolver, ValueMap configDetailMap, St
 			}
 			LOG.info(String.format("CDNDataAPIServiceImpl :: endpoint ::%s, pagepath ::%s, encodedURL ::%s", endpoint, pagePath, encodedURL));
 			URL url = getURL(endpoint, accountId, encodedURL);
-			StringBuilder content = httpGet(url);
-
-			String response = content.toString();
+			Map<String, Object> responseMap = httpGet(url);
+			String response = responseMap.containsKey(BODY) ? responseMap.get(BODY).toString() : "";
+			String eTag = responseMap.containsKey(Constants.E_TAG) ? responseMap.get(Constants.E_TAG).toString() : "";
+			String eTagNodeValue = configDetailMap.containsKey(Constants.E_TAG) ?  (String) configDetailMap.get(Constants.E_TAG) : StringUtils.EMPTY;		
+			if (eTagNodeValue.equals(eTag)) {return;}
```

Comment:
Just formatting here
```suggestion
                        // ETag hasn't changed do not update the cache.
			if (eTagNodeValue.equals(eTag)) {
			    return;
			}
```